### PR TITLE
Added reactive extensions for UISearchController

### PR DESCRIPTION
**Pull request by sergdort**

### CODE REVIEW

**Comment by kzaher** [comment]
Hi @sergdort ,

if you could add delegate tests for `DelegateProxyTest+UIKit.swift`, I think we could merge this.

**Comment by sergdort** [comment]
Hi, @kzaher , completely forgot about this :) 
Will do!

**Comment by sergdort** [comment]
Hi, @kzaher 
Please correct me if I'm wrong.
I am assuming that DelegateProxyTest+UIKit.swift contains tests for delegates which extents some parent delegate, like `UITableViewDelegate` extends `UIScrollViewDelegate`
But `UISearchControllerDelegate` is kind of "root", it only extends `NSObjectProtocol`

Am I correct?

**Comment by kzaher** [comment]
Hi @sergdort ,

yes, `DelegateProxyTest+UIKit.swift` contains extensions tests.

**Comment by sergdort** [comment]
Hi, @kzaher , I'm not sure if did right :)
Please take a look at my tests :)

File: RxCocoa/iOS/Proxies/RxSearchControllerDelegateProxy.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    : DelegateProxy
+    , DelegateProxyType
+    , UISearchControllerDelegate {
+    
+    /**
+     For more information take a look at `DelegateProxyType`.
+     */
+    public class func setCurrentDelegate(delegate: AnyObject?, toObject object: AnyObject) {
+        let imagePickerController: UISearchController = castOrFatalError(object)
+        imagePickerController.delegate = castOptionalOrFatalError(delegate)
```
Comment (Dwar3xwar) [inline RxCocoa/iOS/Proxies/RxSearchControllerDelegateProxy.swift]: Typo? :smile:

Comment (sergdort) [inline RxCocoa/iOS/Proxies/RxSearchControllerDelegateProxy.swift]: Lol 😊, so sorry

**Comment by sergdort** [comment]
Something went wrong 😱

**Pull request merged**